kbase-extension/static/kbase/js/api/upa.js

```diff
@@ -108,13 +108,65 @@ define([
             return deserial;
         };
 
+        var changeUpaVersion = function(upa, newVersion) {
+            if (!isUpa(upa)) {
+                throw {
+                    error: upa + ' is not a valid upa, so its version cannot be changed!'
+                };
+            }
+            var newUpa = upa.replace(/^(.+\/)(\d+)$/, '$1' + newVersion);
+            return newUpa;
+        };
+
+        var serializeAll = function(upas) {
+            if (typeof upas === 'string') {
+                return serialize(upas);
+            }
+            else if (Array.isArray(upas)) {
+                var serialArr = [];
+                upas.forEach(function(upa) {
+                    serialArr.push(serialize(upa));
+                });
+                return serialArr;
+            }
+            else {
+                var serialSt = {};
+                Object.keys(upas).forEach(function(key) {
+                    serialSt[key] = serializeAll(upas[key]);
+                });
+                return serialSt;
+            }
```

Comment:
Same comment as above, could be:
return upas.reduce( function(acc, key) { acc[key] = serializeAll(upas[key]); return acc }, {} );

...but doesn't need to be.